manifest/manifest.go

```diff
@@ -96,3 +114,29 @@ func (m *Manifest) GetComponent(path string) *Component {
 func (m *Manifest) GetOverlay(path string) *Overlay {
 	return m.overlays[path]
 }
+
+// Validate checks if the manifest is valid. If it is, nil is returned. Otherwise an error is returned.
+func Validate(manifest *util.DynamicYaml) error {
+	reservedNamespaces := map[string]bool{
+		"onepanel":           true,
+		"application-system": true,
+		"cert-manager":       true,
+		"istio-system":       true,
+		"knative-serving":    true,
+		"kube-public":        true,
+		"kube-system":        true,
+	}
+
+	defaultNamespace := manifest.GetValue("application.defaultNamespace")
+	if defaultNamespace == nil {
+		return &ParamsError{Key: "application.defaultNamespace", ErrorType: "missing"}
```

Comment:
> @rushtehrani That might be best. Although, it is possible other params could be missing or blank when they should not be - the messages would be the same.

Would it make sense to then have `ErrorType` and `ResourceName` and then provide a message based on the combination those? And maybe have a catch all condition for each `ErrorType` if the `ResourceName` is not defined. We could also create a separate issue for this and expand later.

Reply:
@rushtehrani 

Yeah, let's expand on this separately. 

I'm completely up for expanding/fleshing this out. The important parts to me are:
* The key that caused the issue
* The value (if applicable) that caused the issue.
* What type of error it is - value missing, value blank, value invalid otherwise (how? reserved name, too many characters, etc)
* Bonus: Line that caused the issue.